src/binary-reader-interpreter.cc

```diff
@@ -136,6 +136,18 @@ class BinaryReaderInterpreter : public BinaryReaderNop {
   wabt::Result OnLocalDeclCount(Index count) override;
   wabt::Result OnLocalDecl(Index decl_index, Index count, Type type) override;
 
+  wabt::Result OnAtomicLoadExpr(Opcode opcode,
+                                uint32_t alignment_log2,
```

Comment:
Hm, I started to do that, but if I do it requires the `BinaryReader` to check that the alignment is natural for the instruction, which is a bit strange since the validator needs to do the same check. Most of the checks there currently just ensure that the binary format structure is OK (and even that may be nice to move up a layer, so wasm-objdump can decide what to do about errors).

Reply:
Makes sense.